res/controllers/midi-components-0.1.js

```diff
@@ -0,0 +1,401 @@
+                obj[name] = 0;
+                return obj;
+            }, {});
+        }
+    }
+
+    /** @typedef {function():void} ButtonActionCallback */
+
+    /**
+     * @callback ButtonEventCallback
+     * @param {Button.Events} event
+     */
+
+    /**
+     * @typedef {ComponentOpts} ButtonOpts
+     * @property {Button.Types} [type=Button.Types.PUSH]
+     * @property {number} [on=127]
+     * @property {number} [off=0]
+     * @property {number} [longPressTimeout=275] Time millisecondsto distinguish a short press
+     *      from a long press. It is recommended to refer to it (as this.longPressTimeout)
+     *      in any Buttons that act differently with short and long presses
+     *      to keep the timeouts uniform
+     * @property {ButtonEventCallback} [onEvent=undefined]
+     * @property {ButtonActionCallback} [onPress=undefined]
+     * @property {ButtonActionCallback} [onLongPress=undefined]
+     * @property {ButtonActionCallback} [onDoublePress=undefined]
+     */
+    class Button extends Component {
+        /** @param {ButtonOpts} opts */
+        constructor({
+            type = Button.Types.PUSH,
+            on = 127,
+            off = 0,
+            longPressTimeout = 275,
+            onEvent = undefined,
+            onPress = undefined,
+            onLongPress = undefined,
+            onDoublePress = undefined
+        } = {}) {
+            super(arguments[0]);
+
+            this.type = type;
+            this.on = on;
+            this.off = off;
+            this.longPressTimeout = longPressTimeout;
+
+            // Replacing default methods by user specified ones
+            if (onEvent instanceof Function) {
+                this.onEvent = onEvent.bind(this);
+            }
+
+            if (onPress instanceof Function) {
+                this.onPress = onPress.bind(this);
+            }
+
+            if (onLongPress instanceof Function) {
+                this.onLongPress = onLongPress.bind(this);
+            }
+
+            if (onDoublePress instanceof Function) {
+                this.onDoublePress = onDoublePress.bind(this);
+            }
+
+            this.__btnEvts = new InputTimeWindow(this.longPressTimeout, Object.values(Button.Events), evts => {
+                if (evts[Button.Events.PRESS] === 1 && evts[Button.Events.RELEASE] === 1) {
+                    this.onPress();
+                } else if (evts[Button.Events.PRESS] === 1 && evts[Button.Events.RELEASE] === 0) {
+                    this.onLongPress();
+                } else if (evts[Button.Events.PRESS] === 2) {
+                    this.onDoublePress();
+                } else { // Unbound action
+                    console.debug(`Unbound action to ${evts}`);
+                }
+            });
+        }
+
+        // eslint-disable-next-line no-unused-vars
+        isPress(channel, control, value, status) {
+            return value > 0;
+        }
```

Comment:
We prefer the `_` prefix tactic instead. Or do you have some rationale why the `eslint-disable` is better here?
```suggestion
        isPress(_channel, _control, value, _status) {
            return value > 0;
        }
```

Reply:
The problem with the `_` prefix strategy is that it propagates when you override the method using IDEs code gen like Webstorm's. IMO the one `_` strategy is suited for end users of an API but not when designing the API. That a parameter is unused in the base class is not relevant since these methods are designed to be overriden.